uvloop/loop.pyx

```diff
@@ -2060,8 +2061,10 @@ cdef class Loop:
             waiter = self._new_future()
             tr = TCPTransport.new(self, protocol, None, waiter, context)
             try:
+                # Take ownership of the file descriptor
+                sockfd = sock.detach()
                 # libuv will make socket non-blocking
-                tr._open(sock.fileno())
+                tr._open(sockfd)
```

Comment:
The approach looks correct -- but I'm wondering how vanilla asyncio handles the same thing?

Reply:
I think vanilla asyncio has an easier problem in that it can just have python sockets "all the way down", so just let reference counting take care of cleanup, while here we need to manage the disconnect with `libuv` dealing in file descriptors. I am suspecting there is some error handling path where a file descriptor is closed while the python socket object remains alive and not detached, so when it **is** finally closed, it messes up any new socket that happens to have the same file descriptor.
e.g. create socket `s`, call a loop method passing in an explicit socket, <bad error path which will end with `sock.close()`> overlapping with an `.accept`. I think the `.accept` never results in a python socket object being created.

So with the methods accepting `sockets` and other methods that internally work directly in file descriptors can there be a discrepancy?